src/back/glsl/mod.rs

```diff
@@ -955,56 +909,38 @@ impl<'a, W: Write> Writer<'a, W> {
             self.write_storage_access(access)?;
         }
 
-        // Write the storage class
-        // Trailing space is important
-        if let Some(storage_class) = glsl_storage_class(global.space) {
-            write!(self.out, "{} ", storage_class)?;
+        if let Some(storage_qualifier) = glsl_storage_qualifier(global.space) {
+            write!(self.out, "{} ", storage_qualifier)?;
         }
 
-        // If struct is a block we need to write `block_name { members }` where `block_name` must be
-        // unique between blocks and structs so we add `_block_ID` where `ID` is a `IdGenerator`
-        // generated number so it's unique and `members` are the same as in a struct
-
-        // Write the block name, it's just the struct name appended with `_block_ID`
-        let needs_wrapper = if global.space.is_buffer() {
-            let ty_name = &self.names[&NameKey::Type(global.ty)];
-            let block_name = format!(
-                "{}_block_{}{:?}",
-                ty_name,
-                self.block_id.generate(),
-                self.entry_point.stage,
-            );
-            write!(self.out, "{} ", block_name)?;
-            self.reflection_names_globals.insert(handle, block_name);
-
-            match GlobalTypeKind::new(self.module, global.ty) {
-                GlobalTypeKind::WrappedStruct => {
-                    write!(self.out, "{{ ")?;
-                    // Write the type
-                    // `write_type` adds no leading or trailing spaces
-                    self.write_type(global.ty)?;
-                    true
-                }
-                GlobalTypeKind::Unsized(members) => {
-                    self.write_struct_body(global.ty, members)?;
-                    false
-                }
-                GlobalTypeKind::Other => {
-                    return Err(Error::Custom("Non-struct type of a buffer".to_string()));
-                }
+        match global.space {
+            crate::AddressSpace::Private => {
+                self.write_simple_global(handle, global)?;
             }
-        } else {
-            self.write_type(global.ty)?;
-            false
-        };
-
-        if let crate::AddressSpace::PushConstant = global.space {
-            let global_name = self.get_global_name(handle, global);
-            self.reflection_names_globals.insert(handle, global_name);
+            crate::AddressSpace::WorkGroup => {
+                self.write_simple_global(handle, global)?;
+            }
+            crate::AddressSpace::PushConstant => {
+                self.write_simple_global(handle, global)?;
+            }
+            crate::AddressSpace::Uniform | crate::AddressSpace::Handle => {
+                self.write_interface_block(handle, global)?;
```

Comment:
Isn't `AddressSpace::Handle` supposed to be used for textures and samplers? In that case using `write_interface_block` doesn't make much sense

Reply:
You're right - I'm not sure what's going on here. It didn't change any of our snapshots, but I don't understand why not. I'll figure this out and clarify the code.

Reply:
Indeed, textures and samplers are handled directly by `Writer::write_global`'s caller, `Writer::write`. I've changed `write_global` to refuse to, uh, handle handles.